nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java

```diff
@@ -602,6 +599,28 @@ private WebAppContext loadWar(final File warFile, final String contextPath, fina
         return webappContext;
     }
 
+    private void addHTTPHeaders(WebAppContext webappContext) {
+        // Add a filter to set the X-Frame-Options header
+        FilterHolder xfoFilter = new FilterHolder(new XFrameOptionsFilter());
```

Comment:
I think we had a discussion on this on a prior PR but I don't recall -- what is the advantage of a `FilterHolder` around the filter?

Reply:
I believe the only reason I did this was so that I could set the name for the holder/filter, which made it easier to debug when I was having issues adding filters correctly.

Reply:
I could potentially drop that and do a much simpler add to the webappContext?

Reply:
No no, that makes perfect sense and is evidence that you've actually worked on this, while I just looked at the code. Keep it this way to help with debugging.